server/game/cards/characters/02/will.js

```diff
@@ -0,0 +1,31 @@
+const DrawCard = require('../../../drawcard.js');
+
+class Will extends DrawCard {
+    setupCardAbilities() {
+        this.forcedReaction({
+            when: {
+                onUnopposedWin: (e, challenge) => this.controller !== challenge.winner
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Select and sacrifice a ranger character you control',
+                    waitingPromptTitle: 'Waiting for opponent to use ' + this.name,
+                    cardCondition: card => card.location === 'play area' && card.getType() === 'character' && card.hasTrait('Ranger') && card.controller === this.controller,
+                    onSelect: (p, card) => this.onCardSelected(p, card)
+                });
+            }
+        });
+    }
+
+    onCardSelected(player, card) {
+        player.sacrificeCard(card);
+
+        this.game.addMessage('{0} uses {1} to sacrifice {2}', player, this, card);
```

Comment:
Messages for forced reactions are usually written in the style of 'Player is forced by card to...'.